semeio/jobs/correlated_observations_scaling/job.py

```diff
@@ -1,232 +1,118 @@
-                else:
-                    new_entries.append(val)
-            new_dict[main_key]["keys"] = new_entries
-
-    return new_dict
-
-
-def setup_configuration(input_data, schema):
-    """
-    Creates a ConfigSuite instance and inserts default values
-    """
-    default_layer = job_config.get_default_values()
-    config = configsuite.ConfigSuite(input_data, schema, layers=(default_layer,))
-    return config
-
-
-def _create_active_lists(enkf_observations, events):
-    """
-    Will add observation vectors to observation data. Returns
-    a list of tuples mirroring the user config but also containing
-    the active list where the scaling factor will be applied.
-    """
-    new_events = []
-    observation_data = LocalObsdata("some_name", enkf_observations)
-    for event in events:
-        observation_data.addObsVector(enkf_observations[event.key])
-
-        obs_index = _data_index_to_obs_index(enkf_observations, event.key, event.index)
-        new_active_list = _get_active_list(observation_data, event.key, obs_index)
-
-        new_events.append(_make_tuple(event.key, event.index, new_active_list))
-
-    return new_events
-
-
-def _get_active_list(observation_data, key, index_list):
-    """
-    If the user doesn't supply an index list, the existing active
-    list from the observation is used, otherwise an active list is
-    created from the index list.
-    """
-    if index_list is not None:
-        return _active_list_from_index_list(index_list)
-    else:
-        return observation_data.copy_active_list(key).setParent()
-
-
-def _make_tuple(
-    key,
-    index,
-    active_list,
-    new_event=namedtuple("named_dict", ["key", "index", "active_list"]),
-):
-    return new_event(key, index, active_list)
-
-
-def _update_scaling(obs, scale_factor, events):
-    """
-    Applies the scaling factor to the user specified index, SUMMARY_OBS needs to be treated differently
-    as it only has one data point per node, compared with other observation types which have multiple
-    data points per node.
-    """
-    for event in events:
-        obs_vector = obs[event.key]
-        for index, obs_node in enumerate(obs_vector):
-            if obs_vector.getImplementationType().name == "SUMMARY_OBS":
-                index_list = (
-                    event.index if event.index is not None else range(len(obs_vector))
-                )
-                if index in index_list:
-                    obs_node.set_std_scaling(scale_factor)
-            elif obs_vector.getImplementationType().name != "SUMMARY_OBS":
-                obs_node.updateStdScaling(scale_factor, event.active_list)
-    print(
-        "Keys: {} scaled with scaling factor: {}".format(
-            [event.key for event in events], scale_factor
+from semeio.jobs.correlated_observations_scaling.scaled_matrix import DataMatrix
+from semeio.jobs.correlated_observations_scaling.validator import has_keys, is_subset
+
+
+class ScalingJob(object):
```

Comment:
To me the API of `ScalingJob` and it's usage hints towards a reusable object that can scale different sets of measured data. However, I think the current local assumption of the export location prevents this. Do we intend to open up for using the scaling job multiple times (and if so, does that yield any value?) or is it a create once, use once object?

Reply:
If the scaling job is idempotent (I'm not entirely sure, maybe @oyvindeide can chime in), then there's no reason for why it shouldn't run multiple times.

That the export location might give rise to data overwriting/corrution, shouldn't the concern of the scaling job. Rather, it should be of some parent workflow object.

Reply:
I agree, but then I think the consumer of the ScalingJob should also provide the exporter, as otherwise usage multiple times is very unnatural.

Reply:
Right, this exporter would have to be moved into whatever replaces `ErtScript`. The exporter will exist in `ScalingJob` given `ErtScript`.